File: news/src/main/java/com/leory/badminton/news/mvp/ui/adapter/IndicatorPagerAdapter.java
package com.leory.badminton.news.mvp.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Describe : 指示器adapter
 * Author : leory
 * Date : 2019-05-19
 */
public class IndicatorPagerAdapter extends FragmentPagerAdapter {
    private List<Fragment> fragmentList;

    public IndicatorPagerAdapter(FragmentManager fm, List<Fragment> fragmentList) {
        super(fm);
        this.fragmentList = fragmentList;
    }


    @Override
    public Fragment getItem(int i) {
        return fragmentList.get(i);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }
}

File: news/src/main/java/com/leory/badminton/news/mvp/ui/adapter/IndicatorPagerAdapter.kt
package com.leory.badminton.news.mvp.ui.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Describe : 指示器adapter
 * Author : leory
 * Date : 2019-05-19
 */
class IndicatorPagerAdapter(fm: FragmentManager, private val fragmentList: List<Fragment>) : FragmentPagerAdapter(fm) {


    override fun getItem(i: Int): Fragment {
        return fragmentList[i]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }
}
